packages/message-predicates/contract-message-predicate/script_asm.rs

```diff
@@ -1,65 +1,58 @@
-
-    /* The following assembly code is intended to do the following:
-     *  - Call the function `process_message` on the contract with ID that matches
-     *   the first 32 bytes in the message data field, while forwarding the exact
-     *   amount of base asset specified in the `InputMessage` `amount` field
-     *
-     * note: this code makes the assumption that all memory at VM initialization is set to 0
+    const REF_DATA_START_PTR: u16 = 13 * BYTES_PER_INSTR;
+    
+    /* The following assembly code is intended to
+     * Call the function `process_message` on the contract with ID that matches
+     * the first 32 bytes in the message data field. It won't forward the possible value
+     * stored in the message. L1 entities sending messages here MUST NOT attach
+     * a base asset amount, or it will be permanently lost.
      */
     let mut script: Vec<u8> = vec![
-        //extend stack for contract call data
         op::move_(REG_MEMORY_START_PTR, RegId::SP), //REG_MEMORY_START_PTR = stack pointer
         op::cfei(32 + 32 + 8 + 8), //extends current call frame stack by 32+32+8+8 bytes [base asset id, contract id, param1, param2]
         op::addi(REG_DATA_PTR, REG_MEMORY_START_PTR, 32), //REG_DATA_PTR = REG_MEMORY_START_PTR + 32bytes [memory start pointer + 32]
-        op::addi(REG_DATA_FN_SEL_PTR, REG_DATA_PTR, 32 + 4), //REG_DATA_FN_SEL_PTR = REG_DATA_PTR + 32bytes + 4bytes [call data start pointer + 32 + 4]
-        //prep call parameters
-        op::gtf(
-            REG_MSG_AMOUNT,
-            RegId::ZERO,
-            GTFArgs::InputMessageAmount.into(),
-        ), //REG_MSG_AMOUNT = amount value of message from input[0]
         op::gtf(
             REG_CONTRACT_ADDR_PTR,
             RegId::ZERO,
             GTFArgs::InputMessageData.into(),
         ), //REG_CONTRACT_ADDR_PTR = memory location of the message data from input[0]
-        op::addi(REG_FN_SELECTOR_PTR, RegId::IS, REF_DATA_START_PTR), //REG_FN_SELECTOR_PTR = function selector at end of program
-        op::mcpi(REG_DATA_PTR, REG_CONTRACT_ADDR_PTR, 32), //32 bytes at REG_DATA_PTR = the 32 bytes at REG_CONTRACT_ADDR_PTR
-        op::mcpi(REG_DATA_FN_SEL_PTR, REG_FN_SELECTOR_PTR, 4), //4 bytes at REG_DATA_FN_SEL_PTR = the 4 bytes at REG_FN_SELECTOR_PTR
-        //make contract call
-        op::call(REG_DATA_PTR, REG_MSG_AMOUNT, REG_ASSET_PTR, RegId::CGAS),
+        op::mcpi(REG_DATA_PTR, REG_CONTRACT_ADDR_PTR, 32), // REG_DATA[0..31] = REG_CONTRACT_ADDR_PTR[0..32]
+        op::addi(REG_FN_SELECTOR_PTR, RegId::IS, REF_DATA_START_PTR),
+        op::addi(REG_DATA_FN_SELECTOR_PTR, REG_DATA_PTR, 32), // REG_DATA_FN_SELECTOR_PTR = REG_DATA_PTR + 32
+        op::sw(REG_DATA_FN_SELECTOR_PTR, REG_FN_SELECTOR_PTR, 0), // REG_DATA[32..39] = (End of IS)[0..7] = (len of "process_message")
+        op::addi(REG_CALLDATA_PTR, RegId::IS, REF_DATA_START_PTR + 23), // REG_DATA_FN_SELECTOR_PTR = REG_DATA_PTR + 32 + 23
```

Comment:
Can you replace the hardcoded `23` with `FN_SEL_BYTES_LEN `?

Reply:
Done